eclipse-theia/theia
# feat(vscode): add support for "workbench.extensions.installExtension"

#### What it does

Adds support for the `workbench.extensions.installExtension` command (part of #8387).

We want this because we're considering to use https://github.com/joelspadin-garmin/vscode-private-extension-manager which depends on this API to install extensions.

#### How to test

1. Build the Example app
1. Put the [test extension](https://github.com/fgreinacher/Extension-Commands/releases/tag/3.0.0) in the `plugins` directory
1. Execute the command "Install extension via ID and VS Code API" with a valid an extension ID (e.g. `redhat.java`).
1. Execute the command "Install extension via VSIX and VS Code API" with the URI to a locally available VSIX file (something like `file:///home/gitpod/temp/some-other.vsix`)
1. Ensure that both extensions have been installed via the extension manager.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
